packages/astro/src/vite-plugin-astro-server/plugin.ts

```diff
@@ -51,13 +54,33 @@ export default function createVitePluginAstroServer({
 			const localStorage = new AsyncLocalStorage();
 
 			/** rebuild the route cache + manifest */
-			async function rebuildManifest() {
+			async function rebuildManifest(path: string | null = null) {
 				pipeline.clearRouteCache();
-				routeManifest = injectDefaultDevRoutes(
-					settings,
-					devSSRManifest,
-					await createRouteManifest({ settings, fsMod }, logger), // TODO: Handle partial updates to the manifest
-				);
+
+				// If a route changes, we check if it's part of the manifest and check for its prerender value
+				if (path !== null) {
+					const route = routeManifest.routes.find((r) => path.endsWith(r.component));
+					if (!route) {
+						return;
+					}
```

Comment:
There's an edge case where this causes a conflict:
- Component `src/pages/index.astro` exists
- Component `src/src/pages/index.astro` is edited

`path.endsWith(r.component)` will match the index page and apply the prerender option from `src/src/pages/index.astro` instead of its own.

At first, I thought of suggesting `path === join(settings.root, r.component)`, but that would lead to problems with entry points on `node_modules` using import paths.

Separately, is `path` normalized across platforms to account for `\` and `/` separators?

Reply:
I don't know if it's normalized

> Given Ema's comment that these are already resolved to file paths then they could be compared for equality combining with the root path.

What would you check exactly?

Reply:
I used to do [`e837aff` (#12329)](https://github.com/withastro/astro/pull/12329/commits/e837affae21b0061722ab511fe5a158125a77afd), it that what you're suggesting as well?

Reply:
Yeah, something like that. A probably premature optimization would be to compute `relative(settings.config.root, path)` outside of the `.find()` call and compare it with `r.component` so it doesn't have to construct a new string for comparison on each iteration. But hey, it's JS, so we'd probably need to polish a lot more bits for this to matter.

Reply:
Alright I'll go back to this impl then